core/include/mutex.h

```diff
@@ -215,6 +269,71 @@ void mutex_unlock(mutex_t *mutex);
  */
 void mutex_unlock_and_sleep(mutex_t *mutex);
 
+/**
+ * @brief   Cancels a call to @ref mutex_lock_cancelable
+ *
+ * @param[in,out]   mc      Mutex cancellation structure referring to the
+ *                          thread calling @ref mutex_lock_cancelable and to
+ *                          the mutex to cancel the operation on
+ *
+ * @note    This function is considered internal. Out of tree users should be
+ *          aware that breaking API changes or removal of this API without
+ *          an deprecation period might happen.
+ *
+ * @pre     @p mc is used to cancel at most one call to
+ *          @ref mutex_lock_cancelable. (You can reinitialize the same memory
+ *          to safely reuse it.)
+ * @warning You ***MUST NOT*** call this function once the thread @p mc refers
+ *          to uses the mutex @p mc refers to again (not counting the call
+ *          to @ref mutex_lock_cancelable @p mc was used in).
```

Comment:
I'm sorry but this is still too confusing to me, does phrasing like this conserve the meaning?

```suggestion
 * @warning You ***MUST NOT*** call this function once the thread referred by @p mc
 *          re-uses the mutex object referred by @p mc (not counting the call
 *          to @ref mutex_lock_cancelable @p mc was used in).
```

Reply:
Applied and squashed in